website/client/src/components/inventory/items/index.vue

```diff
@@ -458,12 +465,22 @@ export default {
           }
         });
 
-        itemsArray.sort((a, b) => {
-          if (this.sortBy === 'quantity') {
-            return b.quantity - a.quantity;
-          } // AZ
-          return a.text.localeCompare(b.text);
-        });
+        if (this.sortBy === 'quantity') {
+          const quantitySnapshot = this.quantitySnapshot[groupKey];
+          if (quantitySnapshot) {
+            itemsArray.sort((a, b) => quantitySnapshot[b.key] - quantitySnapshot[a.key]);
+          } else {
+            itemsArray.sort((a, b) => b.quantity - a.quantity);
+          }
+
+          if (quantitySnapshot === null) {
+            this.quantitySnapshot[groupKey] = Object.fromEntries(
+              itemsArray.map(item => [item.key, item.quantity]),
+            );
+          }
```

Comment:
You can avoid the initial branching if you invert the order
```suggestion
          const quantitySnapshot = this.quantitySnapshot[groupKey] || Object.fromEntries(
            itemsArray.map(item => [item.key, item.quantity]),
          );
          itemsArray.sort((a, b) => quantitySnapshot[b.key] - quantitySnapshot[a.key]);
          this.quantitySnapshot[groupKey] = quantitySnapshot;
```